javaparser-core/src/main/java/com/github/javaparser/printer/concretesyntaxmodel/CsmConditional.java

```diff
@@ -93,6 +94,17 @@ boolean evaluate(Node node, ObservableProperty property) {
                 return property.getValueAsBooleanAttribute(node);
             }
         }
+        , IS_STMT_GROUP_ENTRY {
```

Comment:
There may be a more re-usable way to add this, but I couldn't think of anything low-effort enough to justify the effort. Suggestions are very welcome though :)

Reply:
To answer this and the below comment:

This is the condition I used to distinguish the `case x: ...` and `case x -> ...` options for switch statements and expressions respectively (please see my comment above for a better discussion of how it's used).

To explain what it's doing line-by-line:
```
 if (node instanceof SwitchEntry && property == ObservableProperty.TYPE) {
```
This check only makes sense to do for the `TYPE` property of `SwitchEntry` nodes, so check for that here.

```
SwitchEntry.Type entryType = (SwitchEntry.Type)(property.getRawValue(node));
```
Get the `type` from the `SwitchEntry` here. The `getRawValue` call and the cast are both safe because of the check above. We could also just do something like `((SwitchEntry) node).getType()` here, but doing it via the property is more consistent with how the other conditions work.

```
return entryType == SwitchEntry.Type.STATEMENT_GROUP;
```
This check is effectively the whole purpose of this condition. If the entry type is ` STATEMENT_GROUP`, then the entry is a classic switch statement entry `case X: ...`

Reply:
My comment about a more re-usable way to do this was about this condition being very specifically written for this specific check. It would be much nicer to have a general "does the given property on the given node have this value" sort of check, but couldn't think of a good way to do this. Maybe an anonymous class overriding some `expectedValue` could work, but I didn't want to spend too much time engineering a general solution when there's only one use case at this point.

Reply:
> It would be much nicer to have a general "does the given property on the given node have this value" sort of check

It seems to me that this is what the FLAG condition does.

Reply:
I think I answered that in my last comment above, but please let me know if anything is still unclear.